test/solvers/functional/default_functionals_test.py

```diff
@@ -0,0 +1,319 @@
+    space = odl.rn(n)
+    func = odl.solvers.L2NormSquared(space)
+    x = noise_element(space)
+
+    # Evaluation of the functional
+    expected_result = np.sum(x**2)
+    assert almost_equal(func(x), expected_result, places=PLACES)
+
+    # The gradient
+    expected_result = 2.0 * x
+    assert all_almost_equal(func.gradient(x), expected_result,
+                            places=PLACES)
+
+    # The proximal operator
+    sigma = np.random.rand()
+    # Explicit computation: x / (1 + 2*sigma)
+    expected_result = x / (1 + 2.0 * sigma)
+    assert all_almost_equal(func.proximal(sigma)(x), expected_result,
+                            places=PLACES)
+
+    # The convex conjugate functional
+    cc_func = func.convex_conj
+
+    # Evaluation of convex conjugate
+    # Explicit calculation: ||x||^2 / 4
+    expected_result = x.norm()**2 / 4.0
+    assert almost_equal(cc_func(x), expected_result, places=PLACES)
+
+    # The gradient of the convex conjugate (not implemeted)
+    # Explicit calculation: x / 2
+    expected_result = x / 2.0
+    assert all_almost_equal(cc_func.gradient(x), expected_result,
+                            places=PLACES)
+
+    # The proximal of the convex conjugate
+    # Explicit computation: x / (1 + sigma/2)
+    expected_result = x / (1 + sigma / 2.0)
+    assert all_almost_equal(cc_func.proximal(sigma)(x), expected_result,
+                            places=PLACES)
+
+    # The biconjugate, which is the functional itself since it is proper,
+    # convex and lower-semicontinuous
+    cc_cc_func = cc_func.convex_conj
+
+    # Modulo scaling back and forth, check that it is in fact squared L2-norm
+    assert isinstance(cc_cc_func.orig_cconj_f.operator,
+                      odl.solvers.L2NormSquared)
+
+    # Check that they evaluate the same
+    assert almost_equal(cc_cc_func(x), func(x), places=PLACES)
+
+    # Check that they evaluate the gradient the same
+    assert all_almost_equal(cc_cc_func.gradient(x), func.gradient(x),
+                            places=PLACES)
+
+
+def test_constant_functional():
+    """Test the constant functional."""
+
+    n = 10
+    space = odl.rn(n)
+    constant = np.random.randn()
+    func = odl.solvers.ConstantFunctional(space, constant=constant)
+    x = noise_element(space)
+
+    # Checking that constant is stored correctly
+    assert func.constant == constant
+
+    # Evaluation of the functional
+    expected_result = constant
+    assert almost_equal(func(x), expected_result, places=PLACES)
+
+    # The gradient
+    # Given by the zero-operator
+    assert isinstance(func.gradient, odl.ZeroOperator)
+
+    # The proximal operator
+    sigma = np.random.rand()
+    # This is the identity operator
+    assert isinstance(func.proximal(sigma), odl.IdentityOperator)
+
+    # The convex conjugate functional
+    cc_func = func.convex_conj
+
+    # Evaluation of convex conjugate
+    # Explicit calculation: -constant if x=0, infty else
+    if not x.norm() == 0:
```

Comment:
This will never happen by chance, you have to create a explicit test case to trigger it and see that it works.